sycl/test-e2e/Regression/acos.cpp

```diff
@@ -0,0 +1,37 @@
+// REQUIRES: aspect-fp64
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+#include <complex>
+#include <sycl/detail/core.hpp>
+#include <sycl/usm.hpp>
+
+using namespace sycl;
+using T = std::complex<double>;
+
+int main() {
+  queue q;
+  const double pi = std::atan2(+0., -0.);
+  T data[][2] = {{{-2, -INFINITY}, {pi / 2, INFINITY}},
+                 {{-0., INFINITY}, {pi / 2, -INFINITY}}};
+  int N = std::size(data);
+  auto *p = malloc_shared<T>(N, q);
+
+  q.single_task([=] {
+     for (int i = 0; i < N; ++i) {
+       p[i] = std::acos(data[i][0]);
```

Comment:
Note: we do not guarantee that `std::acos` works in kernels. But we should be supporting `::cacos` according to our [docs](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/C-CXX-StandardLibrary.rst).

Reply:
Interesting, it seems we are still testing the std::complex math functions in our testing though: https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/DeviceLib/std_complex_math_fp64_test.cpp